tests/integration/static_ip_address_test.yaml

```diff
@@ -0,0 +1,463 @@
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+description: Configure the ethernet device eth1 to have the address 2001:db8:2::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    address:
+    - ip: 2001:db8:2::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with the address 2001:db8:1::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with the address 192.0.2.251/24 and 2001:db8:1::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the ethernet device eth1 to have the address 192.0.2.251/24 and 192.0.2.252/24
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.252
+      prefix-length: 24
+    - ip: 192.0.2.251
+      prefix-length: 24
+description: Configure the ethernet device eth1 to have the address 2001:db8:1::1/64 and 2001:db8:2::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:2::1
+      prefix-length: 64
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the ethernet device eth1 with empty address
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    address: []
+  ipv6:
+    address: []
+description: Configure the eth1 ethernet device with the address 192.0.2.251/24 and 2001:db8:1::1/64
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+  ipv6:
+    enabled: true
+    dhcp: false
+    autoconf: false
+    address:
+    - ip: 2001:db8:1::1
+      prefix-length: 64
+description: Configure the eth1 ethernet device with the address 192.0.2.251/24
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: false
+    address:
+    - ip: 192.0.2.251
+      prefix-length: 24
+description: Configure the eth1 ethernet device with IPv4 disabled
+interfaces:
+- name: eth1
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: false
+description: Delete the veth terface test-veth1
+interfaces:
+- name: test-veth1
+  type: ethernet
+  state: absent
+description: Configure the veth interface test-veth1 with the peer test-veth1.ep, configure the address 192.0.2.251/24 and address 2001:db8:1::1/64
```

Comment:
`ethernet` interface. This might because nmstate internally only use `ethernet` interface type even it has `veth` section, the `veth` interface type only exists in query state, for apply, they are all unified into `ethernet`.